f5lbaasdriver/v2/bigip/driver_v2.py

```diff
@@ -644,6 +644,72 @@ def refresh(self, context, body):
             self._handle_entity_error(context, loadbalancer.id)
             raise e
 
+    @log_helpers.log_method_call
+    def purge(self, context, body):
+        """Purge a loadbalancer from device without touching DB data."""
+
+        lbext = body["loadbalancerext"]
+        loadbalancer = lbext["loadbalancer"]
+
+        self._log_entity(loadbalancer)
+        driver = self.driver
+        try:
+            agent, device = self._schedule_agent_and_device(context,
+                                                            loadbalancer)
+
+            service = self._create_service(context, loadbalancer, agent)
+            service["device"] = device
+
+            if lbext.get("device", None):
+                device_id_passed = lbext["device"]
+                if device.get("id") != device_id_passed:
+                    LOG.warning('device passed from args. Trying to use it!')
+                    device_loaded = self.driver.device_scheduler.load_device(
+                        context, device_id_passed
+                    )
+
+                    if device_loaded and device_loaded["admin_state_up"]:
```

Comment:
do we need to consider the admin state down device?

Reply:
Currently, admin_state_up == False can prevent cloud user to manipulate configuration against the device. However, admin operations, such as rebuild or purge, might need to manipulate the device when admin set this property False. We may need to consider that a little more.

Reply:
https://github.com/F5Networks/f5-openstack-lbaasv2-driver/pull/1231 is an alternative way which does not check whether admin_state_up or down for device.